src/Libraries/Microsoft.Extensions.AI.Evaluation.NLP/BLEU/MatchCounter.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Shared.Diagnostics;
+
+namespace Microsoft.Extensions.AI.Evaluation.NLP.BLEU;
+
+internal readonly struct MatchCounter<T>
+    where T : IEquatable<T>
+{
+    private readonly Dictionary<T, int> _counts = [];
+
+    public readonly IEnumerable<KeyValuePair<T, int>> Values => _counts;
+
+    public readonly int Sum => _counts.Values.Sum();
```

Comment:
Since the value won't change consider changing as follows -
 
```suggestion
    public readonly int Sum { get; } = _counts.Values.Sum();
```

Reply:
Ah never mind - looks like the value can change.